src/simplewallet/simplewallet.cpp

```diff
@@ -5845,6 +5866,12 @@ bool simple_wallet::transfer_main(Transfer transfer_type, const std::vector<std:
     return false;
 
   std::vector<std::string> local_args = args_;
+
+  static constexpr auto BURN_PREFIX = "burn="sv;
+  uint64_t burn_amount = 0;
+  std::string burn_amount_str = eat_named_argument(local_args, BURN_PREFIX);
+  tools::parse_int(burn_amount_str, burn_amount);
```

Comment:
This will fail if there's a non-integer value here; I think this will work:
```C++
if (!cryptonote::parse_amount(amount, burn_amount_str)) {
  fail_msg_writer() << tr("Invalid amount");
  return true;
}
```
and then also when you pass it into `construct_params(...)` remove the `*COIN` because parse_amount already returns atomic units.

Reply:
I think it probably also needs to check for empty because I don't think parse_amount likes an empty string.
```C++
if (!burn_amount_str.empty() && !cryptonote::parse_amount(burn_amount, burn_amount_str) {
  fail_msg_writer() << tr("Invalid amount");
  return true;
}
```

Reply:
Pushed a commit changing this.